src/openzaak/components/documenten/api/validators.py

```diff
@@ -66,3 +76,71 @@ class UniekeIdentificatieValidator(_UniekeIdentificatieValidator):
 
     def __init__(self):
         super().__init__("bronorganisatie", "identificatie")
+
+
+class RemoteRelationValidator:
+    message = _(
+        "The canonical remote relation still exists, this relation cannot be deleted."
+    )
+    code = "remote-relation-exists"
+
+    def __init__(self, request=None):
+        self.request = request
+
+    def __call__(self, oio: ObjectInformatieObject):
+        # external object
+        if isinstance(oio.object, ProxyMixin):
+            invalid = self._check_remote(oio)
+        else:
+            invalid = self._check_local(oio)
```

Comment:
This works, but I've been trying lately to handle CMIS objects as 'local' as much as possible. Here in the CMIS case, the OIO is also a `ProxyMixin`, and the `does_zaakinformatieobject_exist` and `does_besluitinformatieobject_exist` can be used. So maybe a possibility would be:
```python
        if isinstance(oio.object, ProxyMixin) and not settings.CMIS_ENABLED:
            invalid = self._check_remote(oio)
        else:
            invalid = self._check_local(oio)
```
And then remove the CMIS case in the `_check_remote()` method?